application-admintools-ui/src/main/resources/AdminTools/Code/HealthCheckResources.xml

```diff
@@ -0,0 +1,85 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.5" reference="AdminTools.Code.HealthCheckResources" locale="">
+  <web>AdminTools.Code</web>
+  <name>HealthCheckResources</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>Main.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>HealthCheckResources</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{template name="job_macros.vm"/}}
+
+{{velocity output="false"}}
+#macro (returnHealthCheckJobStatusJSON)
+  #set ($healthCheckJobId = $services.admintools.getHealthCheckJobId())
+  #set ($healthCheckJobStatus = $services.job.getJobStatus($healthCheckJobId))
+  #set ($healthCheckJobStatusJSON = {
+    'id': $healthCheckJobStatus.request.id,
+    'state': $healthCheckJobStatus.state,
+    'progress': {
+      'offset': $healthCheckJobStatus.progress.offset
+    }
+  })
+  #jsonResponse($healthCheckJobStatusJSON)
+#end
+
+#macro (getHealthCheckJobResources)
+  #set ($jobJSURL = $xwiki.getSkinFile('uicomponents/job/job.js'))
+  #set ($jobId = $services.admintools.getHealthCheckJobId())
+  #set ($healthJobStatus = $services.job.getJobStatus($jobId))
+  {{html clean=false}}
+    &lt;div&gt;
+      &lt;span class="jobJSURL"
+        data-jobJSURL="${jobJSURL}"&gt;
+      &lt;/span&gt;
```

Comment:
I'm not sure why, but without an additional SPAN/DIV inside the parent DIV, the jquery find script will not find the searched element. The data from the page is loaded in time for the find script, but if I introduce the class and attribute inside the parent DIV, it will not find the attribute.

Reply:
You mean that in [here](https://github.com/ChiuchiuSorin/application-admintools/blob/health-check/application-admintools-ui/src/main/resources/AdminTools/Code/AdminToolsJS.xml#L168) you cannot find the div in the DOM? How you wrote the selector?

Reply:
Yes. If I modify the [span](https://github.com/ChiuchiuSorin/application-admintools/blob/health-check/application-admintools-ui/src/main/resources/AdminTools/Code/HealthCheckResources.xml#L60) to this [div](https://up1.xwikisas.com/#cCmjcY_CnuYlb4wdujvqBg) and try to use the selector [on the div](https://up1.xwikisas.com/#C4dc1Je8OFsuBllEI1WOUw) instead of the [span](https://github.com/ChiuchiuSorin/application-admintools/blob/health-check/application-admintools-ui/src/main/resources/AdminTools/Code/AdminToolsJS.xml#L168), it will not see the attribute. I'm not sure what is the cause or what I might be missing.